From 16f7bd94a1528dbb57520cd7a02f07114ea36674 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Fri, 15 Jul 2016 10:22:04 -0700
Subject: [PATCH 729/793] drm: xilinx: Add const assignment to fix compile
 warnings

The const was missing for encoder slave functions and drm_mode_fb_cmd2.

Reported-by: Soren Brinkmann <soren.brinkmann@xilinx.com>
Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/gpu/drm/xilinx/xilinx_drm_connector.c |    6 +++---
 drivers/gpu/drm/xilinx/xilinx_drm_encoder.c   |    6 +++---
 drivers/gpu/drm/xilinx/xilinx_drm_fb.c        |    9 +++++----
 drivers/gpu/drm/xilinx/xilinx_drm_fb.h        |    6 +++---
 4 files changed, 14 insertions(+), 13 deletions(-)

diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_connector.c b/drivers/gpu/drm/xilinx/xilinx_drm_connector.c
index 49f9566..ad76dbc 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_connector.c
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_connector.c
@@ -45,7 +45,7 @@ static int xilinx_drm_connector_get_modes(struct drm_connector *base_connector)
 		to_xilinx_connector(base_connector);
 	struct drm_encoder *encoder = connector->encoder;
 	struct drm_encoder_slave *encoder_slave = to_encoder_slave(encoder);
-	struct drm_encoder_slave_funcs *encoder_sfuncs =
+	const struct drm_encoder_slave_funcs *encoder_sfuncs =
 		encoder_slave->slave_funcs;
 	int count = 0;
 
@@ -63,7 +63,7 @@ static int xilinx_drm_connector_mode_valid(struct drm_connector *base_connector,
 		to_xilinx_connector(base_connector);
 	struct drm_encoder *encoder = connector->encoder;
 	struct drm_encoder_slave *encoder_slave = to_encoder_slave(encoder);
-	struct drm_encoder_slave_funcs *encoder_sfuncs =
+	const struct drm_encoder_slave_funcs *encoder_sfuncs =
 		encoder_slave->slave_funcs;
 	int ret = MODE_OK;
 
@@ -97,7 +97,7 @@ xilinx_drm_connector_detect(struct drm_connector *base_connector, bool force)
 	enum drm_connector_status status = connector_status_unknown;
 	struct drm_encoder *encoder = connector->encoder;
 	struct drm_encoder_slave *encoder_slave = to_encoder_slave(encoder);
-	struct drm_encoder_slave_funcs *encoder_sfuncs =
+	const struct drm_encoder_slave_funcs *encoder_sfuncs =
 		encoder_slave->slave_funcs;
 
 	if (encoder_sfuncs->detect)
diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_encoder.c b/drivers/gpu/drm/xilinx/xilinx_drm_encoder.c
index 21b6b8a..51126dd 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_encoder.c
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_encoder.c
@@ -43,7 +43,7 @@ static void xilinx_drm_encoder_dpms(struct drm_encoder *base_encoder, int dpms)
 {
 	struct xilinx_drm_encoder *encoder;
 	struct drm_encoder_slave *encoder_slave;
-	struct drm_encoder_slave_funcs *encoder_sfuncs;
+	const struct drm_encoder_slave_funcs *encoder_sfuncs;
 
 	encoder_slave = to_encoder_slave(base_encoder);
 	encoder_sfuncs = encoder_slave->slave_funcs;
@@ -66,7 +66,7 @@ xilinx_drm_encoder_mode_fixup(struct drm_encoder *base_encoder,
 			      struct drm_display_mode *adjusted_mode)
 {
 	struct drm_encoder_slave *encoder_slave;
-	struct drm_encoder_slave_funcs *encoder_sfuncs = NULL;
+	const struct drm_encoder_slave_funcs *encoder_sfuncs = NULL;
 	bool ret = true;
 
 	encoder_slave = to_encoder_slave(base_encoder);
@@ -84,7 +84,7 @@ static void xilinx_drm_encoder_mode_set(struct drm_encoder *base_encoder,
 					struct drm_display_mode *adjusted_mode)
 {
 	struct drm_encoder_slave *encoder_slave;
-	struct drm_encoder_slave_funcs *encoder_sfuncs;
+	const struct drm_encoder_slave_funcs *encoder_sfuncs;
 
 	DRM_DEBUG_KMS("h: %d, v: %d\n",
 		      adjusted_mode->hdisplay, adjusted_mode->vdisplay);
diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_fb.c b/drivers/gpu/drm/xilinx/xilinx_drm_fb.c
index 269de67..b6ff889 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_fb.c
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_fb.c
@@ -91,7 +91,8 @@ static struct drm_framebuffer_funcs xilinx_drm_fb_funcs = {
  * Return: a xilinx_drm_fb object, or ERR_PTR.
  */
 static struct xilinx_drm_fb *
-xilinx_drm_fb_alloc(struct drm_device *drm, struct drm_mode_fb_cmd2 *mode_cmd,
+xilinx_drm_fb_alloc(struct drm_device *drm,
+		    const struct drm_mode_fb_cmd2 *mode_cmd,
 		    struct drm_gem_cma_object **obj, unsigned int num_planes)
 {
 	struct xilinx_drm_fb *fb;
@@ -466,9 +467,9 @@ void xilinx_drm_fb_restore_mode(struct drm_fb_helper *fb_helper)
  *
  * Return: a drm_framebuffer object if successful, or ERR_PTR.
  */
-struct drm_framebuffer *xilinx_drm_fb_create(struct drm_device *drm,
-					     struct drm_file *file_priv,
-					     struct drm_mode_fb_cmd2 *mode_cmd)
+struct drm_framebuffer *
+xilinx_drm_fb_create(struct drm_device *drm, struct drm_file *file_priv,
+		     const struct drm_mode_fb_cmd2 *mode_cmd)
 {
 	struct xilinx_drm_fb *fb;
 	struct drm_gem_cma_object *objs[4];
diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_fb.h b/drivers/gpu/drm/xilinx/xilinx_drm_fb.h
index 74b628c..1632ba6 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_fb.h
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_fb.h
@@ -30,9 +30,9 @@ xilinx_drm_fb_init(struct drm_device *drm, unsigned int preferred_bpp,
 void xilinx_drm_fb_fini(struct drm_fb_helper *fb_helper);
 
 void xilinx_drm_fb_restore_mode(struct drm_fb_helper *fb_helper);
-struct drm_framebuffer *xilinx_drm_fb_create(struct drm_device *drm,
-					     struct drm_file *file_priv,
-					     struct drm_mode_fb_cmd2 *mode_cmd);
+struct drm_framebuffer *
+xilinx_drm_fb_create(struct drm_device *drm, struct drm_file *file_priv,
+		     const struct drm_mode_fb_cmd2 *mode_cmd);
 void xilinx_drm_fb_hotplug_event(struct drm_fb_helper *fb_helper);
 void xilinx_drm_fb_set_config(struct drm_fb_helper *fb_helper,
 				struct drm_mode_set *set);
-- 
1.7.5.4

